src/components/BoundingClientAwareComponent/BoundingClientAwareComponent.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+type Rect = ClientRect | DOMRect;
+
+export enum PropertyType {
+  VIEWPORT_HEIGHT_MINUS_TOP
+}
+
+export type Property = {
+  type: PropertyType;
+  margin?: number;
+};
+
+type ComputeOffsetProps = {
+  className: string;
+  maxHeight?: Property;
+  handleBoundingClientRect?(rect: Rect): void;
+};
+
+type ComputeOffsetState = {
+  maxHeight?: string;
+};
+
+export const vhMinusTop = (rect: Rect, offset: number) => `calc(100vh - ${rect.top + offset}px)`;
+
+// Computes the BoundingClientRect of the container, this helps to calculate the remaining height without
+// going further off the screen and without having to fix the value in the code.
+// Note: This does re-compute when there is a change in this component, but external changes are not yet
+// managed, that might require to observe the offsets, for our current use case this seems OK, as the top
+// headers doesn't change in height.
+export class BoundingClientAwareComponent extends React.Component<ComputeOffsetProps, ComputeOffsetState> {
```

Comment:
Isn't this `BoundingClientAwareComponent` a little overcomplicated for showing a scrollable list of components?

Reply:
The plan is to use this component in other places where we are fixing stuff like:
`calc(100vh - hard_coded_number_of_px)`
That tends to break easily.

Plus on this case, the math is not direct, it depends on the size of the tip of the arrow, the size of the "clear all", the input, etc.

Reply:
My 2cents is that we need something like this to consolidate these sorts or calculated bounds.  This is probably as good a time as any to introduce it.  [disclaimer: not an expert]

Reply:
Quite complicated, but in the long run, it may be easier to use this, rather than do the math manually. The complexity needs to be somewhere :)